woocommerce/woocommerce-ios
# Customer search: Fill customer fields on Order screen

Part of: #7741
Closes: https://github.com/woocommerce/woocommerce-ios/issues/7941

### Description
This PR deals with: 
- [x] Shows an empty view when we start a new keyword search, for parity with Android.
- [x] Fill the customer details into the Order details fields once a Customer is selected.
- [x] Adds a new view when no Customer search results are found.
- [x] Ignores searches for non-registered customers.

### Changes

- The `OrderCustomerListView` receives a completion handler with a `Customer` object when a merchant taps in the search button. Then we use this data to fill the `EditOrderAddressForm` viewModel.
- Removes `customerSearchTapped()`, as we're not using it anymore.
- When mapping `SearchResults` to `Customer` objects, we skip mapping those with `id = 0` (unregistered), avoiding requests to `/wc/v3/customers/0` that will always fail.

### Testing instructions

1. Go to Orders > + > Create new Order > + Add Customer details > Tap on the magnifying glass icon.
2. Search for a Customer name, and tap on it.

| | | 
|---|---|
| <img width=400 src="https://user-images.githubusercontent.com/3812076/192192292-26a940ff-353f-4c1b-99ba-54d443c0a91d.png"> | <img width=400 src="https://user-images.githubusercontent.com/3812076/198179154-8f594169-dfef-41e2-ab3d-34be43a0c462.png"> |


3. Confirm that Customer fields are filled (will fill both Shipping/Billing if these are different as well).
4. When searching for non-matching keywords, confirm that the empty search results show up.

| | | 
|---|---|
| <img width=400 src="https://user-images.githubusercontent.com/3812076/198179552-a529b898-70ae-4555-bed4-ed59c487270d.png"> | <img width=400 src="https://user-images.githubusercontent.com/3812076/198178774-fde668c6-d88c-4eb9-ba4a-0b269878dabf.png"> |


---
